apps/android/LiveVideoApp/app/src/main/java/com/twilio/livevideo/app/view/SignInPasscodeFragment.kt

```diff
@@ -29,6 +29,6 @@ class SignInPasscodeFragment : SignInBaseFragment() {
     }
 
     private fun registerSignInPasscodeStringObserver() {
-        viewModel.passcode.observe(viewLifecycleOwner) { viewModel.enablePasscodeContinue(it.isNotEmpty()) }
+        viewModel.passcode.observe(viewLifecycleOwner) { viewModel.enablePasscodeContinue(it.isNotEmpty() && it.length >= 8) }
```

Comment:
Looks like you were able to figure out a lot of the passcode details. Good job given I don't think the internals are documented. A few things:

1. I assume `length >= 8` makes `isNotEmpty` redundant
2. I think the [min length for the passcode is actually 14](https://github.com/twilio/twilio-live-interactive-video/blob/ac978a80113d3a4dcfe210f7f58014c34848d78a/apps/ios/LiveVideo/LiveVideo/Managers/Auth/PasscodeComponents.swift#L14-L16) unless I'm missing something
3. If we are going to verify specific length it may be good to move the min length constant to some centralized file that contains all passcode information.
4. In this situation I think iOS just verifies the string is not empty and that's it. That solution would probably be ok also.

Reply:
Good catch Tim!

1. You right, I will remove `isNotEmpty` condition.
2. Yeah, I know the full passcode is 14 digits, but only the last 8 digits and the actual digits used to map into the URL. So, thats the reason of that `8` limit.
3. Sure, I will move this and some other numbers to be constants. 
4. Yes, iOS only verify if it's not empty, but in this case, I implemented a logic to extract the last 8 digits, but if the entered passcode is less than 8 digits, then it cause error. So, i decided to place that logic/validation in the View/UI.

Reply:
Oh I understand why you were using 8. Yeah named constants would help make that more self documenting.

But I think there is still an issue that could happen since the component of the passcode can actually be variable length. I can explain offline.